xdsl/frontend/python_code_check.py

```diff
@@ -104,11 +113,20 @@ def visit_FunctionDef(self, node: ast.FunctionDef) -> None:
 @dataclass
 class MultipleScopeVisitor(ast.NodeVisitor):
 
-    file: str = field(default=None)
+    _file: str | None = field(default=None)
```

Comment:
Same here, the `__init__` will always set `file`

Reply:
- [ ] Check the files again